### PR TITLE
[do not merge] Refactor heading and image mixins and add portrait theme styles

### PR DESCRIPTION
**Pull request by onishiweb**
Adds new mixins for heading styles:
- `oCardHeaderRight` and
- `oCardHeaderLeft`

Adds a new mixin for portrait theme styles: `oCardThemePortrait`

Renames `oCardThemeLandscapeBase` to `oCardThemeLandscape`

Removes `oCardThemeLandscapeImage` mixin

### CODE REVIEW

**Comment by onishiweb** [comment]
@ironsidevsquincy could you have a look through this PR to see if the mixins I've added will allow what you're aiming to do or whether there's a need for some more specific mixins for the general portrait/landscape styles?

**Comment by j-mes** [comment]
LGTM. With the removal of `oCardThemeLandscapeBase` and with `oCardThemeLandscape` instead does send out a much clearer message that it's the only landscape mixin instead of being one of the different landscape options.

**Comment by ironsidevsquincy** [comment]
@onishiweb looks good, will need to actually use it before knowing for sure (sorry, that's an annoying answer)

**Comment by onishiweb** [comment]
@ironsidevsquincy no worries. At this point it's more seeing whether you can spot something that will definitely not work with what you're planning and whether we can mitigate that quickly. I'll get @AlbertoElias to have a look over it and then release this as a v2 beta as it has some breaking changes.

**Comment by onishiweb** [comment]
@j-mes thanks James. Yeh, it was confusing at first but seemed a good option considering there was a general base layout required. But refactoring the image and heading modifiers has helped and now it's a bit more sensible. Cheers

File: src/scss/_themes.scss [hunk 1]
```diff
@@ -23,12 +23,38 @@
...
+@mixin oCardThemePortrait($modifier: 'portrait') {
+
+	&--#{$modifier} {
+		flex-direction: column;
+		flex-wrap: nowrap;
+		justify-content: flex-start;
+	}
+
+	// Normalise the styles for an image in a portrait card
+	&--#{$modifier} &__image {
```
Comment (AlbertoElias) [inline src/scss/_themes.scss]: Why can you change the portrait modifier but not the image element?

Comment (onishiweb) [inline src/scss/_themes.scss]: It's one of the restrictions at the moment, that while the block class and modifier classes are editable, the element classes are fixed. I'm not sure how to really go about making them modifiable without some unnecessarily complicated mixins

Comment (AlbertoElias) [inline src/scss/_themes.scss]: I would even suggest making the modifiers fixed, just to be consistent and only letting the base class be editable

**Comment by onishiweb** [comment]
@AlbertoElias have updated now. I'm unsure what the best solution is in relation to making components transferable to people's projects. I wonder if it's something we should document (i.e. we shouldn't force a `block` class on anyone, but we _can_ define modifier and element classes)

**Comment by AlbertoElias** [comment]
I'd say the rule is, offer at least option to modify the block, and be flexible as much as possible. But to be consistent, I thought it might be a good idea to not make modifies editable, but I don't have a strong opinion

**Comment by onishiweb** [comment]
Fair enough, I think it's good to be as flexible as possible. But would be good to document what _should_ be aimed for in components. Otherwise we'll have lots of inconsistencies across different components.

**Comment by AlbertoElias** [comment]
Feel free to raise a pr in the spec if you think it needs clarifying :)

This pr looks good!

**Comment by j-mes** [comment]
`Add @import 'o-card/main'; to you main SCSS file.`

The `README.md` is missing 'r' in 'you'

**Comment by onishiweb** [comment]
@AlbertoElias this is ready to review again. It's taken a while to get the new layouts sorted the way I wanted so I'm going to leave this PR open and for review before doing the "concept cards" for myFT in another branch/PR.

@ironsidevsquincy sorry it's taken a while but should be ready to go now on this branch.

File: src/scss/_variables.scss [hunk 2]
```diff
@@ -13,4 +13,12 @@ $o-card-spacing-unit: 10px;
...
 /// Size of image when left aligned
 /// @type {Number}
-$o-card-image-width: 100px;
+$o-card-image-width: 40%;
+
+/// Colour of links and tags on Opinion cards
+/// @type {String}
+$o-card-opinion-colour: #416e92; // This should be put into o-colors
+
+$o-card-headshot-colour: #d0e2e7; // This should be put into o-colors
```
Comment (j-mes) [inline src/scss/_variables.scss]: When will this be put in `o-colors` ? Is it after Design decides on the standardisation of the tints? I'm pondering that myself for a different module.

Comment (onishiweb) [inline src/scss/_variables.scss]: Probably before this gets merged. As you mentioned I need to make sure I know where these are coming from and whether they should be put into `o-colors` or whether they should be standardised.

**Comment by ironsidevsquincy** [comment]
Looks good, a couple of things
- the text in right-aligned image cards should wrap under the image

i.e.

screen shot 2016-04-12 at 11 08 02

not

screen shot 2016-04-12 at 11 12 36
- (trickier) related links should appear under the standfirst for right aligned images, but under the image for bottom aligned images

i.e. 

screen shot 2016-04-12 at 11 13 52

and also

screen shot 2016-04-12 at 11 15 41

not

screen shot 2016-04-12 at 11 14 17

**Comment by onishiweb** [comment]
Thanks @ironsidevsquincy I'll take a look into these this afternoon and see what I can do

**Comment by onishiweb** [comment]
@AlbertoElias provided there's no more changes from Next this is ready to review. 

@ironsidevsquincy let me know if you're happy with the changes.

Cheers

**Comment by AlbertoElias** [comment]
As soon as colors have been sorted out, this looks good!

**Comment by onishiweb** [comment]
@AlbertoElias cheers

@ironsidevsquincy what are the plans with the two blues for the opinion card and headshot image background? Are they going into `o-colors` and if so can we do that soon then this can get merged and released.

**Comment by ironsidevsquincy** [comment]
Didn't really have any plans for them. Have no problem with them going into o-colors if that's what you guys want

**Comment by onishiweb** [comment]
Would probably prefer one of the current `o-colors` to be used to be honest, or some guidance on what/why it should be added to `o-colors` etc

**Comment by ironsidevsquincy** [comment]
right, that I can't help with unfortunately. simon (designer) is on holiday atm, should be back any day, once he is I'll get him to talk to you about colours

**Comment by j-mes** [comment]
@ironsidevsquincy That'll be interesting especially since other modules are going to be updated soon and whenever Design has decided on standardisation of any tints (pink, etc) that can be added to an updated `o-colors`.

**Comment by ironsidevsquincy** [comment]
yep, think we've got a few off-piste colours floating around in next

**Pull request merged**